### ISSUE TITLE
All media playback halts while bucklespring is running

### ISSUE THREAD

**Issue by OoLunar** [zevv/bucklespring#92]
Hello! Wonderful program you have going here. Not only does it make me feel slightly more productive, it annoys the hell out of my family! Two birds with one stone!

In all seriousness, I've noticed that I'm not able to listen to any audio related things. Not with VLC, FireFox or anything alike. It blocks both output and input, with exception to the keyboard sounds. With bucklespring disabled, I can play multiple things at once, like Youtube and VLC.

In order to get bucklespring working, I had to modify the `LD_LIBRARY_PATH` and `LIBRARY_PATH` environment variables located in `~/.bashrc`.

```bash
export LD_LIBRARY_PATH=/usr/local/lib/native_client.amd64.cpu.linux:/usr/local/lib:/usr/lib:/lib:/usr/lib/x86_64-linux-gnu/
export LIBRARY_PATH=/usr/local/lib/native_client.amd64.cpu.linux:/usr/local/lib:/usr/lib:/lib:/usr/lib/x86_64-linux-gnu/
```

I also added the following option to my `/etc/modprobe.d/alsa-base.conf`
```
options snd-hda-intel index=-2
```
Here's the verbose output of me switching from youtube to VLC to doing a few bash related tasks, if you find it helpful: https://hastebin.com/yiqaleyigi.properties

I'm not sure if this is related to bucklespring in particular, or OpenAL. I've noticed that you've pointed quite a bit of previous issues to them, which is understandable.

> It blocks both output and input, with exception to the keyboard sounds.

Good news is, now I can finally get the FBI to stop listening to me singing cringy anime openings! :smiley:

I'll help you in anyway that I can, if you can bear with my humor.

**Comment by zevv** [comment]
Hmm, this indeed seems to be an OpenAL issue. Your openAL is likely configured to talk directly to the alsa hardware device, effectively locking it out for any other users. On a common desktop Linux, you'll likely be using Pulseaudio as your audio backend, so OpenAL needs to be configured to use the `pulse` driver to play nicely with the rest of the system.

**Comment by OoLunar** [comment]
Sounds easy like an easy fix. Do you happen to have any docs you can recommend, or should I go Google up some StackOverflow posts?

**Comment by zevv** [comment]
Try this:

```
drivers = pulse
period_size = 8
periods = 4
```

The `period_size` and `periods` args are not strictly necessary, but help lower the latency.

**Comment by OoLunar** [comment]
I put those in my `~/.alsoftrc`, stopped and restarted the buckle binary. Everything still worked just as before; input and output is blocked except for keyboard sounds.

**Comment by zevv** [comment]
Hm not sure how to go from here. As bucklespring does nothing with alsa or sound itself, but only delegates to OpenAL, I recommend asking for help at the OpenAL community, they should be able to figure out how to get this fixed.

Good luck, let me know if you have a solution so I can put it in the docs!

**Comment by zevv** [comment]
Oh wait, did you put the above in the section `[general]` ?

**Comment by OoLunar** [comment]
Apologies for the delay.

My `~/alsoftrc` was empty before you gave me those settings. I wasn't aware it was supposed to be prefilled with things. Maybe there's another one somewhere in my .configs directory...

Anywho, I'll head on over to the OpenAL community and report back what solution they gave me. Thank you for your support!

**Comment by zevv** [comment]
"> Buckle is having user permission issues accessing `/dev/input/event*` files"

Add yourself to the "input" group should fix that on ubunu/debian.

**Comment by zevv** [comment]
Btw, why are you using the `libinput` method anyway, is the normal X11 setup no working for you?

**Comment by OoLunar** [comment]
Turns out there were multiple issues with what I was doing:
- I was running buckle as root, which caused OpenAL to read root's local config `/root/.alsoftrc` instead of my local config that I was modifying `/home/lunar/.alsoftrc`. A fix for this is to modify `/etc/openal/alsoft.con` or not run the program as the current logged in user instead of root.
- While running Buckle as root, it was searching for a pulseaudio server on root. Apparently pulseaudio runs per user instead of system wide. If you see my last response on https://github.com/kcat/openal-soft/issues/493, there's a solution for running with root. Not the recommended solution, and it didn't fix my particular issue, however it's there if someone needs it.

**Adding my user to the `input` group did fix the permissions issue. Everything is running as intended.**

I currently am running Ubuntu 20.04, which uses Wayland. Even though I have Cinnamon DE installed, I'm not sure if it's the Wayland version or not. I switch DE's often and wanted a general solution no matter which DE I used.

Thank you for your time, and I appreciate the project!
